www/src/main/groovy/com/dao/mydebts/settlement/JoinStrategy.groovy

```diff
@@ -0,0 +1,35 @@
+package com.dao.mydebts.settlement
+
+import com.dao.mydebts.entities.StoredAuditEntry
+import com.dao.mydebts.entities.StoredDebt
+import com.dao.mydebts.repos.StoredAuditEntryRepo
+import com.dao.mydebts.repos.StoredDebtRepo
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Controller
+
+/**
+ * Engine implementation based on simple join of two debts with the same src and dest
+ *
+ * @author Alexander Smolko
+ */
+@Controller
+class JoinStrategy implements SettlementStrategy {
+
+    @Autowired
+    StoredDebtRepo sdRepo;
+
+    @Autowired
+    StoredAuditEntryRepo auditEntryRepo
+
+    @Override
+    boolean relax(StoredDebt debt) {
+        def all = sdRepo.findAllNotSettled()
+        all.find { it.id != debt.id && it.src == debt.src && it.dest == debt.dest }.each {
+            auditEntryRepo.save(new StoredAuditEntry(amount: -debt.amount,
```

Comment:
Shouldn't we also create two audit entries here? One for source debt decrease, second positive, for target debt increase?